packages/components/src/test/instance-filter-builder/InstanceFilterConverter.test.ts

```diff
@@ -679,4 +679,39 @@ describe("PresentationInstanceFilter.toInstanceFilterDefinition", () => {
       expect(expression).to.be.eq(`${propertyAccessor} = "a"`);
     });
   });
+
+  describe("handles passed `filteredClasses`", () => {
+    const testImodel = {} as IModelConnection;
+    const classInfo1 = createTestECClassInfo({ id: "0x1" });
+    const classInfo2 = createTestECClassInfo({ id: "0x2" });
+
+    const filter: PresentationInstanceFilterCondition = {
+      field: createTestPropertiesContentField({ properties: [{ property: createTestPropertyInfo({ classInfo: classInfo1 }) }] }),
+      operator: PropertyFilterRuleOperator.IsNull,
+    };
+
+    it(`returns expression with no classes in it when filteredClasses is undefined`, async () => {
+      const { expression } = await PresentationInstanceFilter.toInstanceFilterDefinition(filter, testImodel);
+
+      expect(expression).to.be.eq("this.PropertyName = NULL");
+    });
+
+    it("returns expression with no classes in it when filteredClasses is an empty array", async () => {
+      const { expression } = await PresentationInstanceFilter.toInstanceFilterDefinition(filter, testImodel, []);
+
+      expect(expression).to.be.eq("this.PropertyName = NULL");
+    });
+
+    it("returns expression appended with additional check for classes when one classInfo is passed in", async () => {
+      const { expression } = await PresentationInstanceFilter.toInstanceFilterDefinition(filter, testImodel, [classInfo1]);
+
+      expect(expression).to.be.eq(`this.PropertyName = NULL AND (this.IsOfClass(${classInfo1.id}))`);
+    });
+
+    it("returns expression appended with additional check for classes when array of multiple classInfo's is passed in", async () => {
+      const { expression } = await PresentationInstanceFilter.toInstanceFilterDefinition(filter, testImodel, [classInfo1, classInfo2]);
+
+      expect(expression).to.be.eq(`this.PropertyName = NULL AND (this.IsOfClass(${classInfo1.id}) OR this.IsOfClass(${classInfo2.id}))`);
```

Comment:
please add a test showing multiple property filters in a group get wrapped with parentheses. just to ensure we don't get this: `A OR B AND (class filter)`